data/templates/get_driveid.js

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env node
+
+'use strict';
+
+var exec = require('child_process').exec;
+
+var cmdDriveWwid = 'ls -l /dev/disk/by-id';
+var cmdVdInfo = 'ls -l /dev/disk/by-path';
+var cmdScsiId = 'lsscsi';
+var options = {
+    timeout: 10000
+};
+
+/**
+ * Parse the Drive WWID output
+ * @param {String} data
+ * @return
+ */
+function parseDriveWwid(idList) {
+    //idList example
+    //lrwxrwxrwx. 1 root root  9 Nov 18 20:30 ata-SATADOM-SV_3SE_20150522AA9992050085 -> ../../sdb
+    var lines = idList.split('\n').map(function(line) {
+        var split = line.split(/\s+/);
+        return [split[8],split[10]].join('->');
+    });
+    //According to SCSI-3 spec, vendor specified logic unit name string is 60
+    var scsiLines = [], sataLines = [], requiredStrLen = 60;
+
+    lines.forEach(function(line){
+        if ( line && !(line.match('part'))){
+            var nameIndex = line.lastIndexOf('/'), idIndex = line.lastIndexOf('->');
+            if (line.indexOf('scsi') === 0) {
+                scsiLines.push([line.slice(nameIndex + 1), line.slice(0, idIndex)]);
+            }
+            else if (line.indexOf('ata') === 0) {
+                sataLines.push([line.slice(nameIndex + 1), line.slice(0, idIndex)]);
+            }
+        }
+    });
+
+    var esxiSata = sataLines.map(function(esxiLine) {
```

Comment:
Could you please add a "TODO" description here, I know the conversion from Linux WWID format to ESXi WWID format is just bases on our experiment on limited hardware, we cannot ensure this works for all hardwares. So the TODO item is to find out the method that is suitable for all hardware.